test/jdk/javax/swing/JFileChooser/4525475/JFileChooserReadOnlyTest.java

```diff
@@ -0,0 +1,138 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.Container;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.swing.Action;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.UIManager;
+import javax.swing.UIManager.LookAndFeelInfo;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4525475
+ * @summary This testcase tests JDK-4525475 bug fix, checks whether JFileChooser
+ *          allows modification to the file-system by way of the "New Folder"
+ *          button or not, ideally a read-only JFileChooser shouldn't allow it.
+ * @run main JFileChooserReadOnlyTest
+ */
+public class JFileChooserReadOnlyTest {
+
+    private static boolean result = true;
+    private static boolean newFolderFound = false;
+
+    public static void main(String[] args) {
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            if (!setLookAndFeel(laf)) {
+                continue;
+            }
+
+            // Test1, Read Only JFileChooser
+            boolean readOnly = true;
+            createAndTestCustomFileChooser(readOnly);
+            System.out.println("Its a Read Only JFileChooser " +
+                               (newFolderFound ? "but it has" :
+                                "and it doesn't have") +
+                               " a New Folder Button found" +
+                               ", So the Test1 " +
+                               (result ? "Passed" : "Failed") + " for " + laf);
+
+            // Test2, Read/Write JFileChooser
+            if (!(laf.contains("Motif") || laf.contains("Aqua"))) {
```

Comment:
Can we update the test to find the correct buttons for these L&Fs as well?

Reply:
For Motif L&F, the 'New Folder' button is not shown anywhere irrespective of what we set in UI defaults, for any platform. But Aqua L&F behaves similar to the native FileChooser, like for 'Open' dialog it will not show 'New Folder' button, but it will show 'New Folder' button for 'Save' dialog.

Reply:
@mrserb As the behaviour is as described above, I thought it would be better to skip Motif and Aqua

Reply:
Could you add a comment explaining why these L&Fs are skipped?

Reply:
> But Aqua L&F behaves similar to the native FileChooser, like for 'Open' dialog it will not show 'New Folder' button, but it will show 'New Folder' button for 'Save' dialog.

Does it mean there's New Folder button if you show Save-style `JFileChooser` ?

Reply:
> > But Aqua L&F behaves similar to the native FileChooser, like for 'Open' dialog it will not show 'New Folder' button, but it will show 'New Folder' button for 'Save' dialog.
> 
> Does it mean there's New Folder button if you show Save-style `JFileChooser` ?

Yes, it's there in Save-style `JFileChooser` in Aqua L&F, both in native as well as Swing's JFileChooser.